### PR TITLE
Smoothing out the constructor

### PR DESCRIPTION
**Pull request by thehabes**

### CODE REVIEW

File: js/deer-record.js [hunk 1]
```diff
@@ -213,11 +216,6 @@ export default class DeerReport {
                     UTILS.broadcast(undefined, DEER.EVENTS.LOADED, elem, obj)
                 }).bind(this))
                 .then(() => elem.click())
-        } else {
```
Comment (cubap) [inline js/deer-record.js:216]: Yes, comment on line:77. When there is no `id` all the fields are made dirty. This code isn't exactly right, but if you delete it all than a virgin form will not save fields that have not been entered. Is that the desired behavior?

```
form.deer                         <- type = Event
    input.deer.text             <- becomes dirty when user types "birthday"
    input.deer.hidden        <- is set to dirty when form loads
    input.multiselect          <- updates the hidden, but is not a deer input
    input.deer.datepicker   <- even if it has a default value, will not annotate unless input event
```

This form, if I just load the page and hit submit, saves a new Event with whatever default the hidden is and nothing else. If I add a label and hit submit, I get the same hidden value and a label, but no date, even if it has been set to a default (unless I programmatically make it dirty).

**Comment by thehabes** [comment]
Yeah, that's true.  I realized it wasn't every really used for something real.  I have moved things around now and tested, I think I got it right.

**Comment by thehabes** [comment]
I will put this lived religion w/ local deer code in it up on dev so you can test and track things.

File: js/deer-record.js [hunk 2]
```diff
@@ -71,10 +71,14 @@ export default class DeerReport {
         this.motivation = elem.getAttribute(DEER.MOTIVATION) // inherited to inputs
         this.type = elem.getAttribute(DEER.TYPE)
         this.inputs = elem.querySelectorAll(DEER.INPUTS.map(s => s + "[" + DEER.KEY + "]").join(","))
+        this.inputs.filter(inpt => { return inpt.hasAttribute(DEER.KEY)} ) //Only consider DEER inputs
```
[the remark this answers is not in the document]
Comment (thehabes) [reply js/deer-record.js]: yep

File: js/deer-record.js [hunk 3]
```diff
@@ -90,7 +94,6 @@ export default class DeerReport {
                             let mapsToAnno = false
                             if (deerKeyValue) {
```
[the remark this answers is not in the document]
Comment (thehabes) [reply js/deer-record.js]: yep

File: js/deer-record.js [hunk 4]
```diff
@@ -71,10 +71,14 @@ export default class DeerReport {
         this.motivation = elem.getAttribute(DEER.MOTIVATION) // inherited to inputs
         this.type = elem.getAttribute(DEER.TYPE)
         this.inputs = elem.querySelectorAll(DEER.INPUTS.map(s => s + "[" + DEER.KEY + "]").join(","))
+        this.inputs.filter(inpt => { return inpt.hasAttribute(DEER.KEY)} ) //Only consider DEER inputs
+        this.inputs.forEach(inpt => {
+           inpt.addEventListener('input', (e) => inpt.$isDirty = true) 
+           if (inpt.type === "hidden") { inpt.$isDirty = true }
```
[the remark this answers is not in the document]
Comment (thehabes) [reply js/deer-record.js]: moved it, bad logic.

File: js/deer-record.js (hunk 1 above)
Comment (thehabes) [reply js/deer-record.js:216]: Still needed to call this scenario out, and did so.

File: js/deer-record.js [hunk 5]
```diff
@@ -70,11 +70,12 @@ export default class DeerReport {
...
         this.type = elem.getAttribute(DEER.TYPE)
-        this.inputs = elem.querySelectorAll(DEER.INPUTS.map(s => s + "[" + DEER.KEY + "]").join(","))
-        this.inputs.filter(inpt => { return inpt.hasAttribute(DEER.KEY)} ) //Only consider DEER inputs
+        this.inputs = Array.from(elem.querySelectorAll(DEER.INPUTS.map(s => s + "[" + DEER.KEY + "]").join(",")))
         this.inputs.forEach(inpt => {
-           inpt.addEventListener('input', (e) => inpt.$isDirty = true) 
-           if (inpt.type === "hidden") { inpt.$isDirty = true }
+           inpt.addEventListener('input', (e) => {
+               inpt.$isDirty = true //Make the input dirty
+               this.$isDirty = true //Make the DeerReport dirty
```
Comment (cubap) [inline js/deer-record.js:77]: This line perhaps is not useful yet.

File: js/deer-record.js [hunk 6]
```diff
@@ -86,118 +87,115 @@ export default class DeerReport {
...
-                                    UTILS.warning("Duplicate input " + DEER.KEY + " attribute value '" + deerKeyValue + "' detected in form.  This input will be ignored upon form submission and only the first instance will be respected.  See duplicate below.", el)
-                                    //Don't skip the input though, let it recieve all warnings and errors per usual in case this happens to be the one the dev means to keep.
-                                }
-                                if (obj.hasOwnProperty(deerKeyValue)) {
-                                    if(obj[deerKeyValue].evidence)el.setAttribute(DEER.EVIDENCE, obj[deerKeyValue].evidence)
-                                    if(obj[deerKeyValue].motivation)el.setAttribute(DEER.MOTIVATION, obj[deerKeyValue].motivation)
-                                    if(obj[deerKeyValue].creator)el.setAttribute(DEER.ATTRIBUTION, obj[deerKeyValue].creator)
+                            let mapsToAnno = false                       
+                            let assertedValue = ""
+                            if (flatKeys.indexOf(deerKeyValue) !== i) {
```
Comment (cubap) [inline js/deer-record.js:96]: looking better

File: js/deer-record.js [hunk 7]
```diff
@@ -86,118 +87,115 @@ export default class DeerReport {
...
-                                                console.log(obj[deerKeyValue])
-                                                assertedValue = ""
+                                                arrayOfValues = UTILS.getArrayFromObj(assertedValue, el)
+                                                assertedValue = UTILS.stringifyArray(arrayOfValues, delim)
                                             }
                                         } else {
-                                            //Then getValue returned an object and could not confirm it was an annotation.  We cannot find a value. 
-                                            console.error("Could not find 'value' in the object body.  See below.")
+                                            //This should have been a string or number.  We do not support whatever was meant to be here.  
+                                            console.error("We do not support annotation body values that are objects, unless they are a supported container object and the element " + el.outerHTML + " notes '" + DEER.INPUTTYPE + "'.  Therefore, the value of annotation is being ignored.  See annotation below.")
```
Comment (cubap) [inline js/deer-record.js:177]: helpful

**Pull request merged**